src/dymaptic.Blazor.GIS.API.Core/wwwroot/arcGisJsInterop.js

```diff
@@ -203,45 +195,47 @@ export function buildMapView(dotNetReference, long, lat, rotation, mapObject, zo
                     });
 
                     dotNetReference.invokeMethodAsync('OnViewRendered');
-                    unsetWaitCursor();
+                    unsetWaitCursor(id);
                 } catch (error) {
-                    logError(error);
+                    logError(error, id);
                 }
             });
     } catch (error) {
-        logError(error);
+        logError(error, id);
     }
 }
 
-export function updateWidgets(newWidgets) {
+export function disposeView(viewId) {
     try {
-        setWaitCursor();
-        let oldWidgets = [];
-        activeWidgets.forEach(aw => {
-            if (newWidgets.find(nw => nw.type === aw.type) === undefined) {
-                oldWidgets.push(aw);
-            }
-        });
-        oldWidgets.forEach(ow => {
-            view.ui.remove(ow);
-            activeWidgets.splice(activeWidgets.indexOf(ow), 1);
-            ow?.destroy();
-        });
+        Object.values(arcGisObjectRefs).forEach(o => o?.destroy());
```

Comment:
This takes in a viewId, but doesn't seem to filter `arcGisObjectRefs` with that Id, instead, it destroys everything, is that correct?